File: common-rule-engine/src/test/files/checks/FunctionNameConventionCheck/FunctionNameConventionCheck.java
class Bad_Method_Name {
  public Bad_Method_Name() {
  }

  void bad_name() { // Noncompliant {{Rename 'bad_name' to match the regular expression ^[a-zA-Z0-9]+$.}}
//     ^^^^^^^^
  }

  void good() {
  }

  // FP
  @Override
  void bad_but_overrides(){ // Noncompliant
  }

  @Deprecated
  void bad_2() { // Noncompliant
  }

}

File: common-rule-engine/src/test/files/checks/FunctionNameConventionCheck/FunctionNameConventionCheck.go
package FunctionName

func Do_Something() { // Noncompliant {{Rename 'Do_Something' to match the regular expression ^[a-zA-Z0-9]+$.}}
}

func DoSomething() {
}

func doSomething() {
}

type MyStruct struct {
	X, Y int
}

func (v MyStruct) Method1() int {
	return v.X + v.Y
}

func (v MyStruct) method2() int {
	return v.X + v.Y
}

func (v MyStruct) method_3() int { // Noncompliant
	return v.X + v.Y
}
